Gems/Atom/RHI/Vulkan/Code/CMakeLists.txt

```
@@ -200,6 +103,78 @@ if(PAL_TRAIT_BUILD_HOST_TOOLS)
                 Gem::Atom_RHI_Vulkan.Reflect
                 Gem::Atom_RHI_Vulkan.Builders.Static
                 Gem::Atom_RHI.Edit
+        )
+
+endif()
+
+if(NOT PAL_TRAIT_ATOM_RHI_VULKAN_SUPPORTED)
+
+    # Create stub modules. Once we support gem loading configuration, we can remove this stubbed targets
+    ly_add_target(
+        NAME Atom_RHI_Vulkan.Private ${PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE}
+        NAMESPACE Gem
+        FILES_CMAKE
+            atom_rhi_vulkan_stub_module.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Source
+        BUILD_DEPENDENCIES
+            PRIVATE
+                AZ::AzCore
+                Gem::Atom_RHI.Reflect
     )
```

This can use ly_create_alias to create a target that fufills the requirement for gem loading without the stub
```suggestion
    # Create alias for Atom RHI Vulkan private Gem Module
    ly_create_alias(NAME Atom_RHI_Vulkan.Private NAMESPACE Gem)
```